src/server/models/LogMsg.js

```diff
@@ -0,0 +1,126 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const database = require('./database');
+const { mapToObject, threeDHoleAlgorithm } = require('../util');
+const determineMaxPoints = require('../util/determineMaxPoints');
+const log = require('../log');
+const { isReadonlyKeywordOrPlusOrMinusToken } = require('typescript');
+const LogEmail = require('./LogEmail');
+
+const sqlFile = database.sqlFile;
+
+class LogMsg {
+	/**
+	 * Creates a new log
+	 * @param logType
+	 * @param logMessage
+	 * @param {Moment} logTime
+	 */
+	constructor(logType, logMessage, logTime) {
+		this.logType = logType;
+		this.logMessage = logMessage;
+		this.logTime = logTime;
+	}
+
+	/**
+	 * Creates a new log from data in the row
+	 * @param {*} row The row from which the log is to be created.
+	 * @returns The new log object.
+	 */
+	static mapRow(row) {
+		return new LogMsg(row.logType, row.logMessage, row.logTime);
+	}
+
+	/**
+	 * Returns a promise to create the logging table
+	 * @param conn the database connection to use
+	 * @returns {Promise.<>}
+	 */
+	static createTable(conn) {
+		return conn.none(sqlFile('logmsg/create_logmsg_table.sql'));
+	}
+
+	/**
+	 * Returns a promise to create the logMsgType enum.
+	 * @param {*} conn The connection to use.
+	 * @returns {Promise.<>}
+	 */
+		static createLogMsgTypeEnum(conn) {
+			return conn.none(sqlFile('logmsg/create_log_types_enum.sql'));
+		}
+
+	/**
+	 * Returns a promise to insert this log into the database
+	 * @param conn the database connection to use
+	 * @returns {Promise.<>}
+	 */
+	async insert(conn) {
+		const logMsg = this;
+		await conn.none(sqlFile('logmsg/insert_new_log.sql'), logMsg);
```

Comment:
Just a note that during testing it needs to be tested if the moment object correctly goes into DB value (and also comes out okay in the other functions to get).